presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/JsonRenderer.java

```diff
@@ -30,18 +37,29 @@
 public class JsonRenderer
         implements Renderer<String>
 {
-    private static final JsonCodec<JsonRenderedNode> CODEC = JsonCodec.jsonCodec(JsonRenderedNode.class);
-    private static final JsonCodec<Map<PlanFragmentId, JsonPlanFragment>> PLAN_MAP_CODEC = JsonCodec.mapJsonCodec(PlanFragmentId.class, JsonPlanFragment.class);
+    private static JsonCodec<JsonRenderedNode> codec = JsonCodec.jsonCodec(JsonRenderedNode.class);
+    private static JsonCodec<Map<PlanFragmentId, JsonPlanFragment>> planMapCodec = JsonCodec.mapJsonCodec(PlanFragmentId.class, JsonPlanFragment.class);
+
+    public JsonRenderer(FunctionAndTypeManager functionAndTypeManager)
+    {
+        JsonObjectMapperProvider provider = new JsonObjectMapperProvider();
+        provider.setJsonSerializers(ImmutableMap.of(VariableReferenceExpression.class, new Serialization.VariableReferenceExpressionSerializer()));
+        provider.setKeyDeserializers(ImmutableMap.of(VariableReferenceExpression.class, new Serialization.VariableReferenceExpressionDeserializer(functionAndTypeManager)));
+
+        JsonCodecFactory codecFactory = new JsonCodecFactory(provider, true);
+        this.codec = codecFactory.jsonCodec(JsonRenderedNode.class);
+        this.planMapCodec = codecFactory.mapJsonCodec(PlanFragmentId.class, JsonPlanFragment.class);
+    }
```

Comment:
IMO we have sufficient complexity in this class now that it deserves it's own unit test demonstrating what the JSON output would be for a small distributed/logical plan

Reply:
make sense, add a new test coverage by `TestJsonRenderer`